ruby/ql/lib/codeql/ruby/experimental/Rbi.qll

```diff
@@ -313,19 +311,64 @@ module Rbi {
     MethodSignatureCall getMethodSignatureCall() { result = sigCall }
   }
 
+  bindingset[paramName]
+  private predicate isBlockParamName(string paramName) { paramName = ["blk", "block"] }
+
   /**
    * A call to `params`. This defines the types of parameters to a method or proc.
    */
   class ParamsCall extends MethodCall {
     ParamsCall() { this.getMethodName() = "params" }
 
     /**
-     * Gets the type of a parameter defined by this call.
+     * Gets the type of a positional parameter defined by this call.
      */
-    ParameterType getAParameterType() { result = this.getArgument(_) }
+    ParameterType getPositionalParameterType(int i) {
+      result = this.getArgument(i) and
+      // explicitly exclude keyword parameters
+      not this.getAssociatedParameter(result.getName()) instanceof KeywordParameter and
+      // and exclude block arguments
+      not isBlockParamName(result.getName())
```

Comment:
Shouldn't this be `not result instanceof BlockArgument` instead? Or is using `blk` and `block` the way that RBI models block arguments and parameters?

Reply:
Good point - I think in this case we have something like:
```
  sig do
    params(
      a: String,
      block: T.nilable(T.proc.params(b: Integer).void)
    ).void
  end
  def blk_nilable_method(a, &block); end
```
so the argument here is `block: T.nilable(T.proc.params(b: Integer).void)` which is not a `BlockArgument`, but the associated parameter is `&block` so we can check if that is a `BlockParameter` instead which seems neater and less fragile.